src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/WriteYamlStep.java

```diff
@@ -0,0 +1,187 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package org.jenkinsci.plugins.pipeline.utility.steps.conf;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.model.TaskListener;
+import org.jenkinsci.plugins.pipeline.utility.steps.shaded.org.yaml.snakeyaml.DumperOptions;
+import org.jenkinsci.plugins.pipeline.utility.steps.shaded.org.yaml.snakeyaml.Yaml;
+import org.jenkinsci.plugins.workflow.steps.*;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+import javax.inject.Inject;
+import java.io.*;
+import java.net.URL;
+import java.nio.file.FileAlreadyExistsException;
+import java.util.*;
+
+import static org.apache.commons.lang.StringUtils.isBlank;
+
+/**
+ * Writes a yaml file from the workspace.
+ *
+ * @author Javier DELGADO &lt;witokondoria@gmail.com&gt;.
+ */
+public class WriteYamlStep extends AbstractStepImpl {
+
+    private String file;
+    private Object data;
+
+    @DataBoundConstructor
+    public WriteYamlStep(@Nonnull String file, @Nonnull Object data) {
+        if (file == null) {
+            throw new IllegalArgumentException("file parameter must be provided to writeYaml");
+        }
+        this.file = file;
+        if (data == null) {
+            throw new IllegalArgumentException("data parameter must be provided to writeYaml");
+        } else if (!isValidObjectType(data)) {
+            throw new IllegalArgumentException("data parameter has invalid content (no-basic classes)");
+        }
+        this.data = data;
+    }
+
+    /**
+     * Name of the yaml file to write.
+     *
+     * @return file name
+     */
+    public String getFile() {
+        return file;
+    }
+
+    /**
+     * Name of the yaml file to write.
+     *
+     * @param file file name
+     */
+    public void setFile(String file) {
+        this.file = file;
+    }
+
+    /**
+     * An Object containing data to be saved.
+     *
+     * @return data to save as yaml
+     */
+    public Object getData() {
+        return data;
+    }
+
+    /**
+     * An Object representing the elements to write.
+     *
+     * @param data to parse
+     */
+    public void setData(Object data) {
+        this.data = data;
+    }
+
+    private boolean isValidObjectType(Object obj) {
+        if ((obj instanceof Boolean) || (obj instanceof Character) ||
+            (obj instanceof Number) || (obj instanceof String) ||
+            (obj instanceof URL) || (obj instanceof Calendar) ||
```

Comment:
I had some concerns about this, but it seems it's safe enough:
https://bitbucket.org/asomov/snakeyaml/src/6c6516c003eb4902a0c4c308100c26b99e2ef99c/src/main/java/org/yaml/snakeyaml/representer/SafeRepresenter.java?at=default&fileviewer=file-view-default#SafeRepresenter.java-331